lib/double_entry/locking.rb

```diff
@@ -41,6 +41,13 @@ def self.lock_accounts(*accounts)
       else
         lock.perform_lock(&Proc.new)
       end
+
+    rescue ActiveRecord::StatementInvalid => exception
+      if exception.message =~ /lock wait timeout/i
```

Comment:
Maybe consider using [activerecord's exception translations](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L859-L868) instead? That way people can patch in the postgres equivalent, etc.